doc/api.md

```diff
@@ -0,0 +1,180 @@
+# API
+
+## Endpoint
+
+POST /api/generate-report
+
+## Popis
+
+Endpoint pro generování reportu.
+
+## Parametry
+
+### Query parametry
+
+| Parametr     | Typ    | Povinné | Default | Popis                                                                   |
+|--------------|--------|---------|---------|-------------------------------------------------------------------------|
+| **format**   | string | ne      | pdf     | Formát výsledného reportu. Podporováno: `pdf`, `html`, `typst`.         |
+| **filename** | string | ne      | report  | Název výsledného souboru bez přípony. Přípona se doplní podle `format`. |
```

Comment:
To je jen kvuli tomu response headeru `Content-Disposition`? Jinak to predpokladam nema zadny vyznam.